lib/ReactViews/Map/Panels/SharePanel/SharePanel.tsx

```diff
@@ -0,0 +1,667 @@
+"use strict";
+
+import classNames from "classnames";
+import { observer } from "mobx-react";
+import React, { ChangeEvent, MouseEventHandler } from "react";
+import { Trans, WithTranslation, withTranslation } from "react-i18next";
+import defined from "terriajs-cesium/Source/Core/defined";
+import Clipboard from "../../../Clipboard";
+import Icon from "../../../../Styled/Icon";
+import Loader from "../../../Loader";
+import MenuPanelAlias from "../../../StandardUserInterface/customizable/MenuPanel";
+import Input from "../../../../Styled/Input";
+import DropdownStyles from "../panel.scss";
+import {
+  buildShareLink,
+  buildShortShareLink,
+  canShorten,
+  isShareable
+} from "./BuildShareLink";
+import Styles from "./share-panel.scss";
+import StorySharePanelAlias from "./StorySharePanel";
+import {
+  Category,
+  ShareAction
+} from "../../../../Core/AnalyticEvents/analyticEvents";
+
+import { downloadImg } from "./Print/PrintView";
+import { IReactionDisposer, reaction } from "mobx";
+import Checkbox from "../../../../Styled/Checkbox";
+import Text from "../../../../Styled/Text";
+import Terria from "../../../../Models/Terria";
+import ViewState from "../../../../ReactViewModels/ViewState";
+import { TFunction } from "i18next";
+import CatalogMemberMixin from "../../../../ModelMixins/CatalogMemberMixin";
+
+// Handle incorrect type checking
+const MenuPanel = MenuPanelAlias as any;
+const StorySharePanel = StorySharePanelAlias as any;
+
+interface PropTypes extends WithTranslation {
+  terria: Terria;
+  shortenUrls?: boolean;
+  storyShare: boolean;
+  catalogShare?: boolean;
+  catalogShareWithoutText?: boolean;
+  modalWidth: number;
+  viewState: ViewState;
+  onUserClick: () => void;
+  btnDisabled: boolean;
+  t: TFunction;
+}
+
+interface SharePanelState {
+  isOpen: boolean;
+  // shortenUrls: string | boolean | null;  // TODO: Not being used? Check.
+  shareUrl: string;
+  isDownloading: boolean;
+  advancedIsOpen: boolean;
+  placeholder: string | undefined;
+  errorMessage: string | undefined;
+  includeStoryInShare: boolean;
+}
+
+@observer
+class SharePanel extends React.Component<PropTypes, SharePanelState> {
+  static displayName = "SharePanel";
+  private _unsubscribeFromPrintMediaChange!: () => void;
+  private _oldPrint!: (() => void) & (() => void);
```

Comment:
I don't think non-null assertion is necessary - as all usage of `_oldPrint` and `_unsubscribeFromPrintMediaChange` are type guarded

```suggestion
  private _unsubscribeFromPrintMediaChange?: () => void;
  private _oldPrint?: () => void;
```

Reply:
Implemented cheers